Makefile

```diff
@@ -14,10 +14,10 @@ go: init
 	GO111MODULE=on go build ./pkg/...
 
 rust: init
-	cargo check --features regenerate
+	@cargo check --features regenerate 2>&1 | tee /dev/fd/2 | grep -qvz warning || (echo Please fix warnings; exit 1)
```

Comment:
Could use `RUSTFLAGS="-Dwarnings" cargo check`, rather than grep here.

Reply:
This is exactly what I have tried at the beginning. But turns out that the warning is output by cargo shell rather than rust, so the flag will not work.

Reply:
Ah, of course. Could you add a comment please so we remember that in the future?